### DISCUSSION TITLE
libuv.org DNS is broken

### DISCUSSION THREAD

**Discussion by vcunat** [libuv/libuv#3440] [General]
libuv.org is currently a CNAME, which is violation of standards and commonly causes issues (the name not resolving).  Trace: https://dnsviz.net/d/libuv.org/Ye_Qxw/dnssec/

You can find lots of resources on apex CNAME usage, e.g. BIND: https://www.isc.org/blogs/cname-at-the-apex-of-a-zone/

(_aside: I believe new BIND versions actually use libuv for the packets_)

**Comment by bnoordhuis** [comment]
@saghul You maintain the domain name?

**Reply by saghul** [reply]
👋  Yep I do. I'll take a look.

**Reply by vcunat** [reply]
@saghul: ping, I guess?  I still see the issue.

**Reply by vcunat** [reply]
@saghul: ping, I guess? I still see the issue.  _Hmm, maybe I should just give up._

**Reply by saghul** [reply]
Sorry, I've been a tad busy lately and forgot.

**Reply by saghul** [reply]
Alright I took a look and looks like there is a bit of a mess :-/ I do have the domain, but DNS didn't get migrated, oops! So one thing we could do is set it up from scratch at the registrar site (currently GoDaddy), since we have very few entries. AFAICT:

- libuv.org -> currently the CNAME to GitHub for GH pages, not sure how we should set that up.
- docs.libuv.org -> CNAME to RTD
- dist.libuv.org -> 138.197.224.240

Thoughts?

**Reply by vcunat** [reply]
If it should be on GH pages: https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain

**Reply by saghul** [reply]
👍

**Comment by saghul** [comment]
Sorry it took this long, but this is now fixed:

```
dig libuv.org @ns61.domaincontrol.com

; <<>> DiG 9.10.6 <<>> libuv.org @ns61.domaincontrol.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 31608
;; flags: qr aa rd; QUERY: 1, ANSWER: 4, AUTHORITY: 2, ADDITIONAL: 1
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1472
;; QUESTION SECTION:
;libuv.org.			IN	A

;; ANSWER SECTION:
libuv.org.		3600	IN	A	185.199.109.153
libuv.org.		3600	IN	A	185.199.110.153
libuv.org.		3600	IN	A	185.199.111.153
libuv.org.		3600	IN	A	185.199.108.153

;; AUTHORITY SECTION:
libuv.org.		3600	IN	NS	ns62.domaincontrol.com.
libuv.org.		3600	IN	NS	ns61.domaincontrol.com.

;; Query time: 22 msec
;; SERVER: 97.74.101.32#53(97.74.101.32)
;; WHEN: Tue Jul 11 11:33:38 CEST 2023
;; MSG SIZE  rcvd: 157
```

Might take a bit to reflect due to TTL.
